private/rules/jvm_import.bzl

```diff
@@ -14,35 +14,28 @@ def _jvm_import_impl(ctx):
         fail("Please only specify one jar to import in the jars attribute.")
 
     injar = ctx.files.jars[0]
-    manifest_update_file = ctx.actions.declare_file(injar.basename + ".target_label_manifest", sibling = injar)
-    ctx.actions.expand_template(
-        template = ctx.file._manifest_template,
-        output = manifest_update_file,
-        substitutions = {
-            "{TARGETLABEL}": "%s" % ctx.label,
-        },
-    )
-
     outjar = ctx.actions.declare_file("processed_" + injar.basename, sibling = injar)
     ctx.actions.run_shell(
-        inputs = [injar, manifest_update_file] + ctx.files._host_javabase,
+        inputs = [injar] + ctx.files._add_jar_manifest_entry,
         outputs = [outjar],
-        command = " && ".join([
-            # Make a copy of the original jar, since `jar(1)` modifies the jar in place.
-            "cp {input_jar} {output_jar}".format(input_jar = injar.path, output_jar = outjar.path),
-            # Set the write bit on the copied jar.
-            "chmod +w {output_jar}".format(output_jar = outjar.path),
+        command = "\n".join([
             # If the jar is signed do not modify the manifest because it will
-            # make the signature invalid. Otherwise append the Target-Label
-            # manifest attribute using `jar umf`
-            "(unzip -l {output_jar} | grep -qE 'META-INF/.*\\.SF') || ({jar} umf {manifest_update_file} {output_jar} > /dev/null 2>&1 || true)".format(
-                jar = "%s/bin/jar" % ctx.attr._host_javabase[java_common.JavaRuntimeInfo].java_home,
-                manifest_update_file = manifest_update_file.path,
-                output_jar = outjar.path,
+            # make the signature invalid.
+            "if unzip -l {injar} | grep -qE 'META-INF/.*\\.SF'; then".format(injar = injar.path),
```

Comment:
This assumes that `unzip` is on the machine, which may not be true on Windows. Maybe delegate this to the tool?

Reply:
We really need to avoid using `unzip` here, since it won't work on Windows. Sadly, it doesn't look like `@bazel_tools//tools/zip:zipper` supports just listing files, but we could use `jar tvf {injar}` here relatively safely.

Reply:
I would rather address that in another PR.  This PR was just meant to replace using jar to add the Target-Label entry to the manifest in a cache friendly manner.  The `unzip` method was here before and as far as I know worked fine on Windows.  Removing it is probably better but I think it's a separate PR.  

Please also keep in mind that this feature is not enabled by default.  People have to opt into the using the stamping functionality with the `--@rules_jvm_external//settings:stamp_manifest` option.

Reply:
`unzip` is also a suggested prereq for using Bazel on Windows: https://docs.bazel.build/versions/master/install-windows.html#installing-compilers-and-language-runtimes

We can add that as a setup step in the README.

Reply:
A follow up PR is fine.